src/output/usage.rs

```diff
@@ -387,12 +387,20 @@ impl<'help, 'app, 'parser> Usage<'help, 'app, 'parser> {
         for a in self.p.required.iter() {
             if let Some(m) = matcher {
                 for aa in self.p.app.unroll_requirements_for_arg(a, m) {
-                    unrolled_reqs.push(aa);
+                    // if we don't check for duplicates here this causes duplicate error messages
+                    // see https://github.com/clap-rs/clap/issues/2770
+                    //
+                    // using a HashSet causes re-ordered usages and breaks existing tests
+                    if !unrolled_reqs.contains(&aa) {
+                        unrolled_reqs.push(aa);
+                    }
```

Comment:
We could use `indexmap::IndexSet` which preserves ordering but the number of reqs is probably small enough that its not worth adding an extra dependency

Reply:
happy to switch to that if preferred

Reply:
Like I said, probably not worth it.  I just didn't want to leave my thought process in my head but write it down in case anyone else wondered about this.

Reply:
Oh, looks like we already depend on `indexmap`.  Helps if I look at our dependencies before speaking :)